sdk/quantum/Azure.Quantum.Jobs/tests/QuantumJobClientTestEnvironment.cs

```diff
@@ -1,12 +1,126 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+using System;
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
 using Azure.Core.TestFramework;
+using Azure.Identity;
+using Newtonsoft.Json;
 
 namespace Azure.Quantum.Jobs.Tests
 {
     public class QuantumJobClientTestEnvironment : TestEnvironment
     {
+        private bool _initialized = false;
+        private readonly Lazy<AzLoginAccessTokenInfo> _azLoginAccessTokenInfo = new Lazy<AzLoginAccessTokenInfo>(QuantumJobClientTestEnvironment.GetAzLoginAccessTokenInfo);
+
         public string WorkspaceName => GetRecordedVariable("WORKSPACE_NAME");
+
+        public string GetRandomId(string idName)
+        {
+            var randomId = Guid.NewGuid().ToString("N");
```

Comment:
Move this out of TestEnv and into your test class so you can use `TestRecording.Random.NewGuid()` instead to get random values when recording and predictable values during playback.